libnmstate/state.py

```diff
@@ -329,6 +339,13 @@ def merge_route_config(self, other_state):
 
         return changed_interface
 
+    def merge_dns_config(self, other_state):
+        """
+        If self does contain none DNS configuration, copy it from other_state
```

Comment:
English: If DNS is not mentioned in the self state, overwrite it with the other DNS entries.

The above is assuming you will indeed update self._state as well as mentioned in the previous comment. But even if you do not, a rephrasing is in order.